### PR TITLE
[aws-lambda] Add missing fields on CognitoUserPoolTriggerEvent

### PR DESCRIPTION
**Pull request by dalen**
See:
https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html#cognito-user-pools-lambda-trigger-syntax-pre-token-generation

The linkParameter doesn't seem to be in the documentation, but it is there on the events.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by simonbuchan — COMMENTED** [review]
Probably should also update the tests?
Also seems a bit lame these are all optional - can any of these be not present? The contents of `claimsOverrideDetails` in particular.

File: types/aws-lambda/index.d.ts [hunk 1]
```diff
@@ -322,6 +323,15 @@ export interface CognitoUserPoolTriggerEvent {
         messageAction?: "SUPPRESS";
         desiredDeliveryMediums?: Array<"EMAIL" | "SMS">;
         forceAliasCreation?: boolean;
+        claimsOverrideDetails?: {
+            claimsToAddOrOverride?: { [key: string]: string };
+            claimsToSuppress?: string[];
+            groupOverrideDetails?: {
```
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts]: Docs has "If you provide an empty or null object in the response, then the groups are suppressed.", so maybe allowing `null` for this?

**Comment by dalen** [comment]
> Probably should also update the tests?

sure, will do.

> Also seems a bit lame these are all optional - can any of these be not present? The contents of `claimsOverrideDetails` in particular.

In their examples they only set some of them, so I think they are all optional.
But we should likely split this whole event by `event.triggerSource` sometime as both valid request and response fields vary by trigger source.

File: types/aws-lambda/index.d.ts (hunk 1 above)
Comment (dalen) [reply types/aws-lambda/index.d.ts]: Good point, fixed.

**Pull request merged**